nilearn/plotting/html_surface.py

```diff
@@ -126,6 +141,131 @@ def one_mesh_info(surf_map, surf_mesh, threshold=None, cmap=cm.cold_hot,
     return info
 
 
+def _matplotlib_cm_to_niivue_cm(cmap):
+    """Convert matplotlib colormap to niivue colormap.
+
+    Parameters
+    ----------
+    cmap_name : str
+        Name of the colormap to convert.
+
+    Returns
+    -------
+    cmap : dict of list
+        Converted colormap, with keys "R", "G", "B", "A", "I".
+    """
+    name = None
+    spec = None
+    reverse = False
+
+    if isinstance(cmap, str):
+        name = cmap
+        if name[-2:] == "_r":
+            name = name[:-2]
+            reverse = True
+
+        if name not in datad:
+            print("Colormap not available 0 " + name)
+            return None
+
+        spec = datad[cmap]
+    elif isinstance(cmap, mpl.colors.Colormap):
+        spec = cmap
+        name = cmap.name
+
+    if (type(spec) is not tuple):
+        print("Colormap not available 1 " + name)
+        print(spec)
+        return None
+
+    n_nodes = len(spec)
+
+    if (type(spec[0][1]) is tuple):
+        print("Colormap not available 2 " + name)
+        return None
+
+    print("Converting {} with {} nodes".format(name, n_nodes))
+
+    js = {"R": [], "G": [], "B": [], "A": [], "I": []}
+    for i in range(n_nodes):
+        js["R"].append(int(255 * spec[i][0]))
+        js["G"].append(int(255 * spec[i][1]))
+        js["B"].append(int(255 * spec[i][2]))
+        idx = int(i / (n_nodes - 1) * 255)
+        js["A"].append(int(0.3 * idx))
+        js["I"].append(idx)
+
+    if reverse:
+        for k in js:
+            js[k] = js[k][::-1]
+
+    return js
+
+
+def one_mesh_info_niivue(
+    surf_map,
+    surf_mesh,
+    threshold=None,
+    bg_map=None,
+    cmap=None,
+    colorbar=None
+):
+    """Build dict for plotting one surface map on a single mesh."""
+    info = {}
+
+    # Create temporary directory
+    with Path(tempfile.mkdtemp()) as output_path:
```

Comment:
I'm also wondering if you have to bother saving the giftis and then reading them from the path to bytes. Can't you just take the gifti objects and use the `to_bytes()` method? For example L228 to L230 would become:
```python
        info["surf_mesh"] = base64.b64encode(
            surf_mesh_gifti.to_bytes()
        ).decode("UTF-8")
```